### PR TITLE
Refactor endpoint for retrieving all temporarily deleted products

### PR DESCRIPTION
**Pull request by kevinkoech357**
​
## Description
Following the resources provided by @zxenonx on standard API naming conventions and best practices, I fixed my endpoint of retrieving all temporarily deleted products by replacing underscores with hyphens. I also updated swagger docs to reflect the same.
​
## Related Issue (Link to linear ticket)

​
## Motivation and Context
Following best API practices.
​
## How Has This Been Tested?
Just an api fix.
​
## Screenshots (if appropriate - Postman, etc):
deleted_products
​

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
​
## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: super_admin_1/products/routes.py [hunk 1]
```diff
@@ -704,11 +704,11 @@ def permanent_delete(user_id, product_id):
 # Define a route to get all temporarily deleted products
 
 # WORKS #TESTED AND DOCUMENTED
-@product.route("/temporarily_deleted_products", methods=["GET"], strict_slashes=False)
+@product.route("/temporarily-deleted-products", methods=["GET"])
```
Comment (zxenonx) [inline super_admin_1/products/routes.py:707]:
```suggestion
@product.route("/products?status=soft-deleted", methods=["GET"])
```

File: super_admin_1/products/routes.py [hunk 2]
```diff
@@ -704,11 +704,11 @@ def permanent_delete(user_id, product_id):
...
 @admin_required(request=request)
 def get_temporarily_deleted_products(user_id):
     """
-    Retrieve temporarily deleted products.
+    Retrieve all temporarily deleted products.
     This endpoint allows super admin users to retrieve a list of products that have been temporarily deleted.
     Returns:
         JSON response with status and message:
```
Comment (zxenonx) [inline super_admin_1/products/routes.py:714]: - type hint missing
- proper docstring missing (Kindly check Google Python Style guide - docstring section)

File: swagger_config.yaml [hunk 3]
```diff
@@ -1227,11 +1227,11 @@ paths:
               message:
                 type: string
 
-  /product/temporarily_deleted_products:
+  /product/temporarily-deleted-products:
```
Comment (zxenonx) [inline swagger_config.yaml:1230]:
```suggestion
  /products...:
```

**Review by zxenonx — CHANGES_REQUESTED** [review]
Thanks @kevinkoech357 
I left some comments

File: super_admin_1/products/routes.py (hunk 2 above)
Comment (kevinkoech357) [reply super_admin_1/products/routes.py:714]: On it.

File: swagger_config.yaml [hunk 4]
```diff
@@ -1227,7 +1227,7 @@ paths:
               message:
                 type: string
 
-  /product/temporarily-deleted-products:
+  /products...:
```
Comment (zxenonx) [inline swagger_config.yaml:1230]: Ah, that's not what I suggested o
Check what how i changed the route....

**Review by zxenonx — CHANGES_REQUESTED** [review]
Please check again
I think there's a misunderstanding

File: swagger_config.yaml (hunk 4 above)
Comment (kevinkoech357) [reply swagger_config.yaml]: @zxenonx Sorry, I've updated swagger.yaml to reflect the newly suggested route. I hope the issue has now been sorted.

File: super_admin_1/products/routes.py (hunk 2 above)
Comment (kevinkoech357) [reply super_admin_1/products/routes.py]: @zxenonx How about now? Any mistakes?

**Comment by kevinkoech357** [comment]
@Freeman-kuch how about this?

**Pull request closed without merging**